src/Processor/ShippingMethodProcessor.php

```diff
@@ -0,0 +1,142 @@
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\Shipping\Model\ShippingCategoryInterface;
+use Sylius\Component\Taxation\Model\TaxCategoryInterface;
+
+final class ShippingMethodProcessor implements ResourceProcessorInterface
+{
+    /** @var FactoryInterface */
+    private $shippingMethodFactory;
+
+    /** @var RepositoryInterface */
+    private $shippingMethodRepository;
+
+    /** @var RepositoryInterface */
+    private $zoneRepository;
+
+    /** @var RepositoryInterface */
+    private $categoryRepository;
+
+    /** @var RepositoryInterface */
+    private $taxCategoryRepository;
+
+    /** @var ObjectManager */
+    private $manager;
+
+    /** @var MetadataValidatorInterface */
+    private $metadataValidator;
+
+    /** @var string[] */
+    private $headerKeys;
+
+    public function __construct(
+        FactoryInterface $shippingMethodFactory,
+        RepositoryInterface $shippingMethodRepository,
+        RepositoryInterface $zoneRepository,
+        RepositoryInterface $categoryRepository,
+        RepositoryInterface $taxCategoryRepository,
+        ObjectManager $manager,
+        MetadataValidatorInterface $metadataValidator,
+        array $headerKeys
+    ) {
+        $this->shippingMethodFactory = $shippingMethodFactory;
+        $this->shippingMethodRepository = $shippingMethodRepository;
+        $this->zoneRepository = $zoneRepository;
+        $this->categoryRepository = $categoryRepository;
+        $this->taxCategoryRepository = $taxCategoryRepository;
+        $this->manager = $manager;
+        $this->metadataValidator = $metadataValidator;
+        $this->headerKeys = $headerKeys;
+    }
+
+    public function process(array $data): void
+    {
+        $this->metadataValidator->validateHeaders($this->headerKeys, $data);
+
+        /** @var ShippingMethodInterface $shippingMethod */
+        $shippingMethod = $this->getShippingMethod($data['Code']);
+        $shippingMethod->setZone($this->findZone($data['Zone']));
+        $shippingMethod->setCategory($this->findCategory($data['Category']));
+        $shippingMethod->setCalculator($data['Calculator']);
+        $shippingMethod->setEnabled($data['Enabled']);
+        $shippingMethod->setConfiguration($data['Configuration']);
+        $shippingMethod->setPosition($data['Position']);
+        $shippingMethod->setCategoryRequirement($data['CategoryRequirement']);
+        $shippingMethod->setTaxCategory($this->findTaxCategory($data['TaxCategory']));
+
+        foreach ($data['Translations'] as $locale => $translation) {
+            $shippingMethod->setCurrentLocale($locale);
+            $shippingMethod->setFallbackLocale($locale);
+
+            $shippingMethod->setName($translation['Name']);
+            $shippingMethod->setDescription($translation['Description']);
+        }
+
+        $this->manager->flush();
+    }
+
+    private function getShippingMethod(?string $code): ?ShippingMethodInterface
```

Comment:
The code should be required here and it will always return `ShippingMethodInterface`.